e2e/env/env.go

```diff
@@ -86,50 +86,66 @@ func setupClient(kubeconfigPath string) (client.Client, error) {
 }
 
 func New(config *types.Config, log *zap.SugaredLogger) (*types.Env, error) {
-	var err error
-
 	env := &types.Env{}
 
-	env.Hub.Name = config.Clusters["hub"].Name
-	if env.Hub.Name == "" {
-		env.Hub.Name = defaultHubClusterName
-		log.Infof("Cluster \"hub\" name not set, using default name %q", defaultHubClusterName)
+	if err := setupHub(&env.Hub, config.Clusters["hub"], defaultHubClusterName, log); err != nil {
+		return nil, fmt.Errorf("failed to setup hub cluster \"hub\": %w", err)
```

Comment:
Can we create the error inside setupHub? It has the context we add here. The context this function can add is "failed to create env".

If we add the context in setupXXX functions, we don't need to repeat it here like we do for the managed clusters.